src/model.c

```diff
@@ -180,6 +180,38 @@ void model_normalize(struct model *model, bool invert_z)
         model_invert_triangles(model);
 }
 
+void model_reserve_vertexes(struct model *model, unsigned int count)
+{
+    if (model->vertex_capacity < count)
+    {
+        model->vertex_capacity = count;
+        vec3 *new_vertices = (vec3 *)realloc(model->vertexes, sizeof(vec3) * count);
+        if (!new_vertices)
+        {
+            fprintf(stderr, "ERROR: Failed to allocate memory for vertexes.\n");
+            exit(1);
+        }
+
+        model->vertexes = new_vertices;
+    }
+}
+
+void model_reserve_faces(struct model *model, unsigned int count)
+{
+    if (model->faces_capacity < count)
+    {
+        model->faces_capacity = count;
+        struct face *new_faces = (struct face *)realloc(model->faces, sizeof(struct face) * count);
+        if (!new_faces)
+        {
+            fprintf(stderr, "ERROR: Failed to allocate memory for faces.\n");
+            exit(1);
+        }
+
+        model->faces = new_faces;
+    }
+}
+
```

Comment:
My opinion is that keeping the code simple is more valuable that a little more efficiency at load time. I suggest not using these functions, since model_add_vertex() and model_add_face() can handle reallocations.

Reply:
I removed these functions and leave the reallocation to model_add_vertex() and model_add_face().